nixos/modules/system/boot/systemd.nix

```
@@ -629,7 +639,10 @@ in
     # https://github.com/systemd/systemd/pull/12226
     boot.kernel.sysctl."kernel.pid_max" = mkIf pkgs.stdenv.is64bit (lib.mkDefault 4194304);
 
-    boot.kernelParams = optional (!cfg.enableUnifiedCgroupHierarchy) "systemd.unified_cgroup_hierarchy=0";
+    boot.kernelParams = lib.foldr (a: b: a ++ b) [] [
+      (optional (cfg.machineId != null) "systemd.machine_id=${cfg.machineId}")
+      (optional (!cfg.enableUnifiedCgroupHierarchy) "systemd.unified_cgroup_hierarchy=0")
+    ];
```

Nitpick. My brain and nix processes this code much faster (also less parentheses):

```suggestion
    boot.kernelParams = optional (cfg.machineId != null) "systemd.machine_id=${cfg.machineId}"
      ++ optional (!cfg.enableUnifiedCgroupHierarchy) "systemd.unified_cgroup_hierarchy=0";
```